File: api-tester/src/defaults/java/com/revenuecat/apitester/java/RedeemWebPurchaseListenerAPI.java
package com.revenuecat.apitester.java;

import androidx.annotation.OptIn;

import com.revenuecat.purchases.CustomerInfo;
import com.revenuecat.purchases.ExperimentalPreviewRevenueCatPurchasesAPI;
import com.revenuecat.purchases.PurchasesError;
import com.revenuecat.purchases.interfaces.RedeemWebPurchaseListener;

@OptIn(markerClass = ExperimentalPreviewRevenueCatPurchasesAPI.class)
@SuppressWarnings({"unused"})
final class RedeemWebPurchaseListenerAPI {
    static void checkListener(RedeemWebPurchaseListener listener,
                              RedeemWebPurchaseListener.Result result) {
        listener.handleResult(result);
    }

    static void checkRedeemResult(RedeemWebPurchaseListener.Result result) {
        if (result instanceof RedeemWebPurchaseListener.Result.Success) {
            CustomerInfo customerInfo = ((RedeemWebPurchaseListener.Result.Success) result).getCustomerInfo();
        } else if (result instanceof RedeemWebPurchaseListener.Result.Error) {
            PurchasesError error = ((RedeemWebPurchaseListener.Result.Error) result).getError();
        } else if (result instanceof RedeemWebPurchaseListener.Result.InvalidToken) {
            return;
        } else if (result instanceof RedeemWebPurchaseListener.Result.AlreadyRedeemed) {
            return;
        } else if (result instanceof RedeemWebPurchaseListener.Result.Expired) {
            String obfuscatedEmail = ((RedeemWebPurchaseListener.Result.Expired) result).getObfuscatedEmail();
            Boolean wasEmailSent = ((RedeemWebPurchaseListener.Result.Expired) result).getWasEmailSent();
        }

        boolean isSuccess = result.isSuccess();
    }
}

File: api-tester/src/defaults/kotlin/com/revenuecat/apitester/kotlin/RedeemWebPurchaseListenerAPI.kt
package com.revenuecat.apitester.kotlin

import com.revenuecat.purchases.CustomerInfo
import com.revenuecat.purchases.PurchasesError
import com.revenuecat.purchases.interfaces.RedeemWebPurchaseListener

@Suppress("unused", "UNUSED_VARIABLE", "ReturnCount")
private class RedeemWebPurchaseListenerAPI {
    fun checkListener(
        redeemWebPurchaseListener: RedeemWebPurchaseListener,
        result: RedeemWebPurchaseListener.Result,
    ) {
        redeemWebPurchaseListener.handleResult(result)
    }

    fun checkResult(result: RedeemWebPurchaseListener.Result): Boolean {
        val isSuccess: Boolean = result.isSuccess

        when (result) {
            is RedeemWebPurchaseListener.Result.Success -> {
                val customerInfo: CustomerInfo = result.customerInfo
                return true
            }
            is RedeemWebPurchaseListener.Result.Error -> {
                val error: PurchasesError = result.error
                return false
            }
            is RedeemWebPurchaseListener.Result.InvalidToken -> {
                return false
            }
            is RedeemWebPurchaseListener.Result.PurchaseBelongsToOtherUser -> {
                return false
            }
            is RedeemWebPurchaseListener.Result.Expired -> {
                val obfuscatedEmail: String = result.obfuscatedEmail
                return false
            }
        }
    }
}
